src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -614,6 +615,26 @@ CitusCreateDirectory(StringInfo directoryName)
 }
 
 
+#ifdef WIN32
+static bool
+FileIsLink(char *filename, struct stat filestat)
+{
+	DWORD attributes = GetFileAttributes(filename);
+	return attributes & FILE_ATTRIBUTE_REPARSE_POINT;
+}
+
+
+#else
+static bool
+FileIsLink(char *filename, struct stat filestat)
+{
+	return S_ISLNK(filestat.st_mode);
+}
+
+
+#endif
+
+
 /*
  * CitusRemoveDirectory first checks if the given directory exists. If it does, the
```

Comment:
I think we should delete this function and use `rmtree()` instead. They seem to do the same thing.

Reply:
Added some comments to your PR. I think that's a reasonable aim for a future PR but don't know if it belongs in this one.

Reply:
👍